public/js/loginValidation.js

```diff
@@ -1,47 +1,47 @@
-function isValidEmail(elementID) {
-    let inputVal = document.getElementById(elementID);
+let isValidEmail = (elementID) => {
+    let inputVal = document.getElementById(elementID)
+    let userEmailWarning = document.getElementById('userEmailWarning')
     let email = inputVal.value.trim();
-    let re = new RegExp("^\\w+([\\.-]?\\w+)*@\\w+([\\.-]?\\w+)*(\\.\\w{2,3})+$");
-
+    let re = new RegExp("^\\w+([\\.-]?\\w+)*@\\w+([\\.-]?\\w+)*(\\.\\w{2,3})+$")
+console.log(re.test(email))
     if (re.test(email)) {
-        inputVal.style.backgroundColor = "";
+        userEmailWarning.classList.remove('error')
         return true
     } else {
-        inputVal.style.backgroundColor = "pink";
+        userEmailWarning.classList.add('error')
         return false
     }
 }
 
-function postLoginDetails(inputs){
-    fetch("/api/login",
+let validateToPostInputs = () => {
+    let cleanedEmailInput = encodeURI(document.getElementById('userEmail').value)
+    let cleanedPasswordInput = encodeURI(document.getElementById('password').value)
+
+    if (isValidEmail('userEmail')) {
+        let emailPasswordValues = {'email' : cleanedEmailInput, 'password': cleanedPasswordInput}
+        return emailPasswordValues
+    } else {
+        document.getElementById('submitWarning').textContent = 'Field input error'
+    }
+}
+
+let loginDetails = (path, data) => {
+    fetch(`/api/${path}`,
         {
             headers: {
                 'Accept': 'application/json',
                 'Content-Type': 'application/json'
             },
             method: "POST",
-            body: JSON.stringify({
-                "userEmail": inputs['email'],
-                "password": inputs['password']
-            })
+            body: JSON.stringify(data)
         })
         .then(function(data){return data.json()})
 }
 
-function validateToPostInputs() {
-    let cleanedEmailInput = encodeURI(document.getElementById('userEmail').value);
-    let cleanedPasswordInput = encodeURI(document.getElementById('password').value);
-
-    if (isValidEmail('userEmail')) {
-        let emailPasswordValues = {'email' : cleanedEmailInput, 'password': cleanedPasswordInput};
-        return emailPasswordValues;
-    } else {
-        document.getElementById('submitWarning').textContent = 'Field input error'
-    }
-}
-
-document.getElementById('formID').addEventListener('submit', function(e){
-    e.preventDefault();
-    let validInputs = validateToPostInputs();
-    postLoginDetails(validInputs);
-});
\ No newline at end of file
+let loginForm = document.getElementById('formID')
```

Comment:
could we rename 'formID' with something more unique? (just thinking there might be more forms in the file / project and should try to keep IDs unique...

Reply:
renamed to 'loginForm'